compiler/rustc_hir_analysis/src/delegation.rs

```diff
@@ -125,8 +165,13 @@ impl<'tcx> GenericsBuilder<'tcx> {
 
         for (idx, param) in own_params.iter_mut().enumerate() {
             param.index = (idx + parent_count) as u32;
-            // Default type parameters are not inherited: they are not allowed
-            // in fn's.
+            // FIXME(fn_delegation): Default parameters are not inherited, because they are
+            // not permitted in functions. Therefore, there are 2 options here:
+            //
+            // - We can create non-default generic parameters.
+            // - We can substitute default parameters into the signature.
+            //
+            // At the moment, first option has been selected as the most general.
             if let ty::GenericParamDefKind::Type { synthetic, .. } = param.kind {
                 param.kind = ty::GenericParamDefKind::Type { has_default: false, synthetic }
```

Comment:
```suggestion
                *has_default = false;
```
The `GenericParamDefKind::Const` logic was also lost in refactoring.